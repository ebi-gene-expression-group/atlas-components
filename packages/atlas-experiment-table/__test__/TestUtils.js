import React from 'react'
import styled from 'styled-components'

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

const TableCellDiv = styled.div`
  font-size: 13px;
  font-family: Helvetica, Arial, FreeSans, "Liberation Sans", sans-serif;
`

const tableHeader = [
  { type: `sort`, title: `Loaded date`, width: 140, dataParam: `lastUpdate` },
  { type: `search`, title: `species`, width: 200, dataParam: `species` },
  { type: ``, title: `experiment description`, width: 360, dataParam: `experimentDescription`, link: `experimentAccession`, resource: `experiments`, endpoint: `Results` },
  { type: `plain`, title: `experiment factors`, width: 260, dataParam: `experimentalFactors` },
  { type: `sort`, title: `Number of assays`, width: 160, dataParam: `numberOfAssays`, link: `experimentAccession`, resource: `experiments`, endpoint: `Experiment Design` },
]

const data = [
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-EHCA-2`,
    experimentDescription: `Melanoma infiltration`,
    lastUpdate: `16-11-2018`,
    numberOfAssays: 6638,
    numberOfContrasts: 0,
    species: `Mus musculus`,
    kingdom: `animals`,
    experimentalFactors: [`single cell identifier`, `sampling site`,`time`],
  },
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-GEOD-99058`,
    experimentDescription: `Single cell`,
    lastUpdate: `11-10-2018`,
    numberOfAssays: 250,
    numberOfContrasts: 0,
    species: `Mus musculus`,
    kingdom: `animals`,
    experimentalFactors: [`single cell identifier`]
  },
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-MTAB-5061`,
    experimentDescription: `healthy individuals and type 2 diabetes patients`,
    lastUpdate: `11-10-2018`,
    numberOfAssays: 3514,
    numberOfContrasts: 0,
    species: `Homo sapiens`,
    kingdom: `plants`,
    experimentalFactors: [`single cell identifier`, `disease`],
  }
]

export {getRandomInt, TableCellDiv, tableHeader, data}
