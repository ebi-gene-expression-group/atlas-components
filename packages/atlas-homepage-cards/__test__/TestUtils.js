

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

const dummyCards = () => {
  return [
    {
      iconType: `species`,
      iconSrc: `mus musculus`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
    {
      iconType: `species`,
      iconSrc: `homo sapiens`,
      iconDescription: `Homo sapiens`,
      content: [
        {
          text: `2 experiments`,
        }
      ]
    },
    {
      iconType: `unknown`,
      iconSrc: `http://www.lingyun.com`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
    {
      iconType: `experiments`,
      iconSrc: `http://www.zhao.com`,
      iconDescription: `Homo sapiens`,
      content: [
        {
          text: `2 experiments`,
        }
      ]
    },
    {
      iconType: `image`,
      iconSrc: `logo-goes-here`,
      iconDescription: `An image card`,
      content: [
        {
          text: `Go to experiments`,
        }
      ]
    },
    {
      iconType: `imagespecies`,
      iconSrc: `http://www.lingyun.com`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
  ]
}

export {getRandomInt, dummyCards}