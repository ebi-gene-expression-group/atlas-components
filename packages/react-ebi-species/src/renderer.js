"use strict";

//*------------------------------------------------------------------*

var React = require('react');
var ReactDOM = require('react-dom');

//*------------------------------------------------------------------*

var Icon = require('./SpeciesIcon.jsx');

//*------------------------------------------------------------------*

module.exports = function(mountNode,species,colourOverride, colourPerKingdom) {
    ReactDOM.render(
        React.createElement(Icon,{
          species: species,
          colourOverride: colourOverride,
          colourPerKingdom: colourPerKingdom
        }), mountNode
    );
};
