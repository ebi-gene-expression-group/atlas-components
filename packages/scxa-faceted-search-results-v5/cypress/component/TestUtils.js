import facets from '../fixtures/vindicatorsResponse.json'
import organismParts from '../fixtures/organismPartsResponse.json'
import cellTypes from '../fixtures/cellTypesResponse.json'
import species from '../fixtures/speciesResponse.json'

import PropTypes from "prop-types";
import styled from "styled-components";
import React from "react";
import CheckboxFacetGroupsDefaultProps from "../../src/facetgroups/CheckboxFacetGroupsDefaultProps";
import MultiSelectDropdownFacetGroupsDefaultProps
  from "../../src/facetgroups/MultiSelectDropdownFacetGroupsDefaultProps";

const getPropsForCheckBoxGroupWithTooltip = () => {
  return Cypress._.shuffle(CheckboxFacetGroupsDefaultProps).pop()
}

const getPropsForMultiSelectDropdownGroupWithTooltip = () => {
  return Cypress._.shuffle(MultiSelectDropdownFacetGroupsDefaultProps).pop()
}

const getPropsForMarkerGeneFacet = () => {
  const markerGeneName = `Marker genes`;
  return CheckboxFacetGroupsDefaultProps.find(facet => facet.name === markerGeneName)
}

const getPropsWithoutTooltip = () => {
  let props = getPropsForCheckBoxGroupWithTooltip()
  props.description = ``
  return props
}

const getFacets = () => {
  return Cypress._.shuffle(facets).filter(() => Math.random() > 0.5)
}

const getRandomSpecies = () => {
  return Cypress._.shuffle(species).filter(() => Math.random() > 0.5)
}

const getRandomOrganismParts = () => {
  return Cypress._.shuffle(organismParts).filter(() => Math.random() > 0.5)
}

const getRandomCellTypes = () => {
  return Cypress._.shuffle(cellTypes).filter(() => Math.random() > 0.5)
}

const ExperimentTableCard = ({title}) =>
  <div>
    <p>{title}</p>
  </div>

ExperimentTableCard.propTypes = {
  title: PropTypes.string.isRequired
}

const IconDiv = styled.div`
  width: 15%;
  text-align: center;
`
IconDiv.displayName = 'IconDiv'

const BitDiv = styled.div`
  width: 25%;
  text-align: center;
`
const DoveDiv = styled.div`
  width: 25%;
  text-align: center;
`

const ExperimentTableHeader = ({onClick}) =>
  <DoveDiv>
    <IconDiv onClick={()=>onClick(`test`)}>
      Meow
    </IconDiv>
    <BitDiv onClick={()=>onClick(`test`)}>
      Wow wow
    </BitDiv>
  </DoveDiv>

ExperimentTableHeader.propTypes = {
  onClick: PropTypes.func.isRequired
}

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

export { getRandomInt, getFacets, getRandomOrganismParts, getRandomCellTypes, getRandomSpecies,
  getPropsForCheckBoxGroupWithTooltip, getPropsForMultiSelectDropdownGroupWithTooltip,
  getPropsWithoutTooltip, getPropsForMarkerGeneFacet,
  ExperimentTableCard, ExperimentTableHeader }
