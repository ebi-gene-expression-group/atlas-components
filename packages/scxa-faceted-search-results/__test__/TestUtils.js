import React from 'react'
import PropTypes from 'prop-types'

const EpisodeCard = ({title}) =>
  <div>
    <p>{title}</p>
  </div>

EpisodeCard.propTypes = {
  title: PropTypes.string.isRequired
}

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min)) + min; //The maximum is exclusive and the minimum is inclusive
}

const vindicators = [
  { group:`Vindicator`, value: `supernova`, label: `Supernova`},
  { group:`Vindicator`, value: `noob-noob`, label: `Noob-Noob` },
  { group:`Vindicator`, value: `vance_maximus`, label: `Vance Maximus`},
  { group:`Vindicator`, value: `alan_rails`, label: `Alan Rails`},
  { group:`Vindicator`, value: `crocubot`, label: `Crocubot` },
  { group:`Vindicator`, value: `million_ants`, label: `Million Ants`},
  { group:`Vindicator`, value: `morty_smith`, label: `Morty Smith`},
  { group:`Vindicator`, value: `rick_sanchez`, label: `Rick Sanchez`},
  { group:`Vindicator`, value: `lady_katana`, label: `Lady Katana`},
  { group:`Vindicator`, value: `calypso`, label: `Calypso`},
  { group:`Vindicator`, value: `diablo_verde`, label: `Diablo Verde`}
]

const episodes = [
  {
    element: {
      title: "Raising Gazorpazorp"
    },
    facets: [
      {
        group: "Planet",
        value: "gazorpazorp",
        label: "Gazorpazorp"
      },
      {
        group: "Guest character",
        value: "gwendolyn",
        label: "Gwendolyn"
      },
      {
        group: "Guest character",
        value: "ma-sha",
        label: "Ma-Sha"
      },
      {
        group: "Season",
        value: "1",
        label: "1"
      }
    ]
  },
  {
    element: {
      title: "The wedding squanchers"
    },
    facets: [
      {
        group: "Planet",
        value: "squanch",
        label: "Squanch"
      },
      {
        group: "Guest character",
        value: "birdperson",
        label: "Birdperson"
      },
      {
        group: "Guest character",
        value: "squanchy",
        label: "Squanchy"
      },
      {
        group: "Season",
        value: "2",
        label: "2"
      }
    ]
  },
  {
    element: {
      title: "The Rickshank redemption"
    },
    facets: [
      {
        group: "Guest character",
        value: "birdperson",
        label: "Birdperson"
      },
      {
        group: "Planet",
        value: "buttworld",
        label: "Buttworld"
      },
      {
        group: "Season",
        value: "3",
        label: "3"
      }
    ]
  },
  {
    element: {
      title: "Ricksy business"
    },
    facets: [
      {
        group: "Guest character",
        value: "squanchy",
        label: "Squanchy"
      },
      {
        group: "Guest character",
        value: "abradolf_lincler",
        label: "Abradolf Lincler"
      },
      {
        group: "Season",
        value: "1",
        label: "1"
      }
    ]
  },
  {
    element: {
      title: "Close Rick-counters of the Rick kind"
    },
    facets: [
      {
        group: "Guest character",
        value: "ricktiminus_sancheziminius",
        label: "Ricktiminus Sancheziminius"
      },
      {
        group: "Guest character",
        value: "abradolf_lincler",
        label: "Abradolf Lincler"
      },
      {
        group: "Planet",
        value: "buttworld",
        label: "Buttworld"
      },
      {
        group: "Season",
        value: "1",
        label: "1"
      }
    ]
  }
]



export {getRandomInt, vindicators, episodes, EpisodeCard}
