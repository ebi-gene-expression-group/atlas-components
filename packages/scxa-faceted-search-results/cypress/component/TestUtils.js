import episodes from '../fixtures/episodes.json'
import PropTypes from "prop-types";
import styled from "styled-components";
import React from "react";

const facets = () => {
  return episodes.reduce((acc, episode) => acc.concat(episode.facets), [])
}

const getUniqueFacets = () => {
  const allFacets = facets()
  return allFacets
    .filter((facet, index) => allFacets.findIndex((thatFacet) => facet.value === thatFacet.value) === index)
    .map((facet) => ({
      ...facet,
      disabled: false
    }))
}

const getFacetTooltip = () => {
  return Cypress._.shuffle(getUniqueFacets()).find(facet => facet.description)
}

const getFacetWithoutTooltip = () => {
  return Cypress._.shuffle(getUniqueFacets()).find(facet => !facet.description)
}

const getPropsWithTooltip = (facetTooltip) => {
  return {
    facetGroupName: facetTooltip.group,
      facetGroupNameDescription: facetTooltip.description,
    facets: getUniqueFacets().filter(facet => facet.group === facetTooltip.group),
    onChange: () => {}
  }
}

const getPropsWithoutTooltip = () => {
  const facetWithoutTooltip = getFacetWithoutTooltip()
  return {
    facetGroupName: facetWithoutTooltip.group,
      facets: getUniqueFacets().filter(facet => facet.group === facetWithoutTooltip.group),
    onChange: () => {}
  }
}

const ExperimentTableCard = ({title}) =>
  <div>
    <p>{title}</p>
  </div>

ExperimentTableCard.propTypes = {
  title: PropTypes.string.isRequired
}

const IconDiv = styled.div`
  width: 15%;
  text-align: center;
`
IconDiv.displayName = 'IconDiv'

const BitDiv = styled.div`
  width: 25%;
  text-align: center;
`
const DoveDiv = styled.div`
  width: 25%;
  text-align: center;
`

const ExperimentTableHeader = ({onClick}) =>
  <DoveDiv>
    <IconDiv onClick={()=>onClick(`test`)}>
      Meow
    </IconDiv>
    <BitDiv onClick={()=>onClick(`test`)}>
      Wow wow
    </BitDiv>
  </DoveDiv>

ExperimentTableHeader.propTypes = {
  onClick: PropTypes.func.isRequired
}

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

export { episodes, getRandomInt, getUniqueFacets, getFacetTooltip, getPropsWithTooltip, getPropsWithoutTooltip,
  ExperimentTableCard, ExperimentTableHeader }
