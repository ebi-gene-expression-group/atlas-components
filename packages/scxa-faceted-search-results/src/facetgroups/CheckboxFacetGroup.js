import React from 'react'
import PropTypes from 'prop-types'
import { xorBy as _xorBy } from 'lodash'

import FacetGroupPropTypes from './FacetGroupPropTypes'
import TooltipIcon from './TooltipIcon'

const CheckboxOption = ({ group, value, label, disabled, checked, onChange }) =>
  <div>
    <input type={`checkbox`} {...{value, checked, disabled}}
      onChange={() => onChange({group, label, value, disabled})}/>
    <label style={disabled ? {color: `lightgrey`} : {}}>{label}</label>
  </div>

CheckboxOption.propTypes = {
  group: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  disabled: PropTypes.bool.isRequired,
  checked: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired
}

// In principle we donâ€™t need this component to be stateful, but in doing so we can create a custom _handleChange
// function that will ultimately call onChange(facetGroupName, facets); this allows us to have the same API as
// React-Select and reuse the same callback for both checkbox-style and multiselect facet groups
class CheckboxFacetGroup extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      checkedFacets: []
    }
    this._handleChange = this._handleChange.bind(this)
  }

  // disabled is passed in but not used (it makes tests easier)
  _handleChange(facet) {
    this.setState(
      { checkedFacets: _xorBy(this.state.checkedFacets, [facet], `value`) },
      () => this.props.onChange(facet.group, this.state.checkedFacets))
  }

  render() {
    const { facetGroupName, facetGroupNameDescription, facets } = this.props
    const { checkedFacets } = this.state

    return (
      <div className={`padding-bottom-xlarge`}>
        <h4>
          { facetGroupName }
          { facetGroupNameDescription && <TooltipIcon tooltipText={facetGroupNameDescription}/> }
        </h4>
        {facets.map((facet) =>
          <CheckboxOption {...facet}
            checked={checkedFacets.some((checkedFacet) => checkedFacet.value === facet.value)}
            onChange={this._handleChange}
            key={facet.value}/>
        )}
      </div>
    )
  }
}

CheckboxFacetGroup.propTypes = FacetGroupPropTypes

export default CheckboxFacetGroup
