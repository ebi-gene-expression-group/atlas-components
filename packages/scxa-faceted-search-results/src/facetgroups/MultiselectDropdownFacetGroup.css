/* Unfortunately we need to set some EBI VF overrides here because it applies styles to elements deper than
React-Selectâ€™s APIs (e.g. the div that encloses the input rather than the input itself) */
.input-clear input, .input-clear input:focus {
  height: unset;
  box-shadow: none;
  margin: 0;
}
