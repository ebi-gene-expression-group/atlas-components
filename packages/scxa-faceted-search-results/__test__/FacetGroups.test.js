import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import {mount} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import {getRandomInt, vindicators} from './TestUtils'

import CheckboxFacetGroup from '../src/facetgroups/CheckboxFacetGroup'
import MultiselectDropdownFacetGroup from '../src/facetgroups/MultiselectDropdownFacetGroup'
import Select from 'react-select'

Enzyme.configure({ adapter: new Adapter() })

const props = {
  facetGroupName: `Vindicators`,
  hideName: false,
  facets: [],
  onChange: () => {}
}

describe(`CheckboxFacetGroup`, () => {
  beforeEach(() => {
    while (props.facets.length === 0) {
      props.facets = vindicators.filter((vindicator) => Math.random() > 0.5)
    }
  })

  test(`renders the right number of checkboxes and the facet name`, () => {
    const wrapper = mount(<CheckboxFacetGroup {...props} />)
    expect(wrapper.find(`input`)).toHaveLength(props.facets.length)
    expect(wrapper.find({ type: `checkbox` })).toHaveLength(props.facets.length)
    expect(wrapper.find(`h4`)).toHaveLength(1)
    expect(wrapper.find(`h4`).text()).toEqual(props.facetGroupName)
  })

  test(`can hide the header`, () => {
    const wrapper = mount(<CheckboxFacetGroup {...props} hideName={true} />)
    expect(wrapper.find(`h4`)).toHaveLength(0)
  })

  test(`callback is called when a checkbox is checked/unchecked with the right arguments`, () => {
    const randomCheckboxIndex = getRandomInt(0, props.facets.length)
    const mockCallback = jest.fn()
    const wrapper = mount(<CheckboxFacetGroup {...props} onChange={mockCallback} />)
    wrapper.find({ type: `checkbox` }).at(randomCheckboxIndex).simulate(`change`)
    expect(mockCallback.mock.calls.length).toBe(1)
    expect(mockCallback.mock.calls[0]).toEqual([props.facets[randomCheckboxIndex].group, [props.facets[randomCheckboxIndex]]])
    wrapper.find({ type: `checkbox` }).at(randomCheckboxIndex).simulate(`change`)
    expect(mockCallback.mock.calls.length).toBe(2)
    expect(mockCallback.mock.calls[1]).toEqual([props.facets[randomCheckboxIndex].group, []])
  })

  test(`matches snapshot`, () => {
    const tree = renderer.create(<CheckboxFacetGroup {...props} facets={vindicators}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

// I would’ve liked to test the MultiselectDropdownFacetGroup component a bit more, for instance some workflow like
// “clicking on the component opens up the menu and choosing an option invokes the callback with the right arguments”
// (similar to CheckboxFacetGroup above) but I tried simulating a click event on a number of components (even the
// dropdown indicator) and nothing happened. It turns out that React-Select isn’t very straightforward when it comes to
// testing, as one can see in e.g. https://github.com/JedWatson/react-select/blob/v2/src/__tests__/Select.test.js.
// Still, a good deal of such a test would end up in testing-the-behaviour-of-third-party-software territory, so below
// is the minimum I think we should at least cover (snapshot tests would be pretty dumb, though). Fortunately coverage
// of React-Select is very good and the people behind the component are using Jest and Enzyme too, so if at any point
// we really want to do some complex testing we can have a look at the repo to learn how to do so... properly.
describe(`MultiselectDropdownFacetGroup`, () => {
  beforeEach(() => {
    while (props.facets.length === 0) {
      props.facets = vindicators.filter((vindicator) => Math.random() > 0.5)
    }
  })

  test(`can hide the header`, () => {
    const wrapper = mount(<CheckboxFacetGroup {...props} hideName={true} />)
    expect(wrapper.find(`h4`)).toHaveLength(0)
  })

  test(`callback includes facet name in arguments`, () => {
    const mockCallback = jest.fn()
    const wrapper = mount(<MultiselectDropdownFacetGroup {...props} onChange={mockCallback} />)
    wrapper.find(Select).instance().onChange()
    expect(mockCallback.mock.calls.length).toBe(1)
    expect(mockCallback.mock.calls[0][0]).toEqual(props.facetGroupName)
  })

})
