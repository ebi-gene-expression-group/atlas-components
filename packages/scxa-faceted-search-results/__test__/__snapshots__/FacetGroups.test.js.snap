// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxFacetGroup matches snapshot 1`] = `
<div
  className="padding-bottom-xlarge"
>
  <h4>
    Vindicators
  </h4>
  <div>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
      value="supernova"
    />
    <label>
      Supernova
    </label>
  </div>
  <div>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
      value="noob-noob"
    />
    <label>
      Noob-Noob
    </label>
  </div>
  <div>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
      value="vance_maximus"
    />
    <label>
      Vance Maximus
    </label>
  </div>
  <div>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
      value="alan_rails"
    />
    <label>
      Alan Rails
    </label>
  </div>
  <div>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
      value="crocubot"
    />
    <label>
      Crocubot
    </label>
  </div>
  <div>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
      value="million_ants"
    />
    <label>
      Million Ants
    </label>
  </div>
  <div>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
      value="morty_smith"
    />
    <label>
      Morty Smith
    </label>
  </div>
  <div>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
      value="rick_sanchez"
    />
    <label>
      Rick Sanchez
    </label>
  </div>
  <div>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
      value="lady_katana"
    />
    <label>
      Lady Katana
    </label>
  </div>
  <div>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
      value="calypso"
    />
    <label>
      Calypso
    </label>
  </div>
  <div>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
      value="diablo_verde"
    />
    <label>
      Diablo Verde
    </label>
  </div>
</div>
`;
